### PR TITLE
Fix in-place obb `gt_bboxes` modification in `RotatedTaskAlignedAssigner`

### LINKED ISSUE #24205: [Performance Regression] YOLO26-OBB shows lower accuracy than YOLO11, and 8.4.x degrades accuracy compared to 8.3.x
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

I have observed two significant performance issues regarding model accuracy in recent experiments:

1. Model Comparison: Under the same training configurations and datasets, the YOLO26 series OBB models show lower accuracy (mAP) compared to the YOLO11/v8 series.
2. Version Regression: Training YOLO11/v8n-obb using ultralytics 8.3.x yields better results than using ultralytics 8.4.x. There seems to be a performance regression in the 8.4.x versions.

Here are the comparative results on the same test set: (only one label & YOLOv8 updated)
| Model | Version | Precision | Recall | mAP50 | mAP50-95 | Notes |
| :---: | :---: | :---: | :---: | :---: | :---: | :---: |
| YOLOv8n | 8.3.253 | 0.9743 | 0.9459 | 0.9819 | 0.9105 | Comparable to 11 |
| YOLOv8n | 8.4.36 | 0.9663 | 0.9502 | 0.9678 | 0.8272 | Drop observed(mAP50-95) |
| YOLO11n | 8.3.253 | 0.9909 | 0.9437 | 0.9801 | 0.9099 | Baseline |
| YOLO11n | 8.4.36 | 0.9753 | 0.9404 | 0.9633 | 0.8374 | Drop observed |
| YOLO26n | 8.4.36 | 0.9456 | 0.9026 | 0.9473 | 0.8182 | Lower than 11 |

### Environment

Ultralytics 8.4.36 🚀 Python-3.11.14 torch-2.9.1+cu128 CUDA:0 (NVIDIA GeForce RTX 3080 Ti, 11911MiB)
Setup complete ✅ (12 CPUs, 62.6 GB RAM, 1988.4/2750.5 GB disk)

OS                     Linux-6.8.0-51-generic-x86_64-with-glibc2.35
Environment            Docker
Python                 3.11.14
Install                pip
Path                   /opt/conda/lib/python3.11/site-packages/ultralytics
RAM                    62.60 GB
Disk                   1988.4/2750.5 GB
CPU                    Intel Core i7-8700K 3.70GHz
CPU count              12
GPU                    NVIDIA GeForce RTX 3080 Ti, 11911MiB
GPU count              1
CUDA                   12.8

numpy                  ✅ 2.2.6>=1.23.0
matplotlib             ✅ 3.10.8>=3.3.0
opencv-python          ✅ 4.12.0.88>=4.6.0
pillow                 ✅ 11.3.0>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.32.5>=2.23.0
scipy                  ✅ 1.17.0>=1.4.1
torch                  ✅ 2.9.1+cu128>=1.8.0
torch                  ✅ 2.9.1+cu128!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.24.1+cu128>=0.9.0
psutil                 ✅ 7.1.3>=5.8.0
polars                 ✅ 1.37.1>=0.20.0
ultralytics-thop       ✅ 2.0.18>=2.0.18

### Minimal Reproducible Example

```
# All model trained on this config

import os
os.environ['YOLO_OFFLINE'] = 'True'

from ultralytics import YOLO, settings

settings.update({"tensorboard": True})

model = YOLO("yolo11n-obb.pt")

results = model.train(
    data="/work/datasets/56_Barcode_OBB/56_Barcode_OBB.yaml",
    project="/work/datasets/56_Barcode_OBB/runs",
    name="train_11n_14",

    epochs=600,
    imgsz=640,
    batch=64,
    device=0,
    workers=6,
    patience=100,

    degrees=180,
    scale=0.7,
    shear=10,
)
```

### Additional

_No response_

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by lmycross**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
🛠️ This PR fixes an in-place modification issue in rotated box target assignment by cloning ground-truth boxes before adjusting very small boxes.

### 📊 Key Changes
- Added `gt_bboxes_clone = gt_bboxes.clone()` in `ultralytics/utils/tal.py` to avoid modifying the original ground-truth tensor directly.
- Updated the small-width/height box handling logic to operate on the cloned tensor instead of `gt_bboxes`.
- Switched the downstream corner conversion call from `gt_bboxes` to `gt_bboxes_clone` when computing rotated box corners.

### 🎯 Purpose & Impact
- Prevents unintended side effects from in-place edits to `gt_bboxes`, which can cause hard-to-track bugs during training ⚠️
- Makes the task-aligned assigner logic safer and more predictable, especially for rotated bounding box workflows 🔄
- Improves code reliability for small-object handling without changing the intended behavior of the model ✅
- Reduces the risk of training instability or incorrect label processing in oriented detection tasks 📦

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lmycross, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review preparation, and an engineer will assist you soon.

To ensure a seamless integration of your work, please review the following checklist:
- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The change looks clean and well-scoped. Cloning `gt_bboxes` before width/height adjustment addresses the unintended in-place mutation without altering the downstream selection flow. I did not find any actionable issues in the provided diff.

**Comment by Laughing-q** [comment]
@lmycross nice spot! awesome work! merging

**Comment by UltralyticsAssistant** [comment]
🎉 Excited to see this merged — thank you @lmycross for the thoughtful fix, and @Laughing-q for the valuable contributions along the way!

As Leonardo da Vinci said, *“Details make perfection, and perfection is not a detail.”* This PR is a great example of that mindset: a small but important safeguard that prevents unintended in-place side effects and makes rotated box target assignment more reliable and predictable.

Really appreciate the care you both put into improving training stability and robustness for oriented detection workflows. Contributions like this make Ultralytics stronger for everyone 🚀
